packages/frontend/src/hooks/gdrive/useGdrive.ts

```diff
@@ -4,11 +4,12 @@ import {
     type ApiScheduledDownloadCsv,
     type UploadMetricGsheet,
 } from '@lightdash/common';
-import { useQuery, type UseQueryOptions } from '@tanstack/react-query';
-import { useState } from 'react';
+import { useMutation, type UseQueryOptions } from '@tanstack/react-query';
+import { useEffect, useRef } from 'react';
 import { lightdashApi } from '../../api';
 import { convertDateFilters } from '../../utils/dateFilter';
 import useHealth from '../health/useHealth';
+import useToaster from '../toaster/useToaster';
 
 const getGdriveAccessToken = async () =>
```

Comment:
Before it had a query with a refetch interval because there was no communication with the auth popup. It would never show errors.

Now it fetches the access token once, if that fails opens the popup and waits for the popup to communicate back with a success of failure state. 
If is it a failure, it shows an error toast. 
If it is a success, it fetches the access token again. If it fails to get a token again, it shows an error toast with more detail on why it failed.